uint8 leg
uint16 a
uint16 b
uint16 c